doc/modules/pipeline.rst

```diff
@@ -124,6 +119,88 @@ i.e. if the last estimator is a classifier, the :class:`Pipeline` can be used
 as a classifier. If the last estimator is a transformer, again, so is the
 pipeline.
 
+Caching transformers: avoid repeated computation
+-------------------------------------------------
+
+.. currentmodule:: sklearn.pipeline
+
+Fitting transformers may sometimes be computational expensive. With its
+``memory`` parameter set, class:`Pipeline` will cache each transformer
+after calling ``fit``.
+This feature is used to avoid computing the fit transformers within a pipeline
+if the parameters and input data are identical. A typical example is the case of
+a grid search in which the transformers can be fitted only once and reuse for
+each configuration.
+
+The parameter ``memory`` is needed in order to cache the transformers.
+``memory`` can be either a string containing the directory where to cache the
+transfomers or a `joblib.Memory <https://pythonhosted.org/joblib/memory.html>`_
+object::
+
+    >>> from tempfile import mkdtemp
+    >>> from shutil import rmtree
+    >>> from sklearn.decomposition import PCA
+    >>> from sklearn.svm import SVC
+    >>> from sklearn.pipeline import Pipeline
+    >>> estimators = [('reduce_dim', PCA()), ('clf', SVC())]
+    >>> cachedir = mkdtemp()
+    >>> pipe = Pipeline(estimators, memory=cachedir)
+    >>> pipe # doctest: +NORMALIZE_WHITESPACE, +ELLIPSIS
+    Pipeline(...,
+             steps=[('reduce_dim', PCA(copy=True,...)),
+                    ('clf', SVC(C=1.0,...))])
+    >>> # Clear the cache directory when you don't need it anymore
+    >>> rmtree(cachedir)
+
+.. warning:: **Side effect of caching transfomers**
+
+   Enabling caching triggers a clone of the transformers before fitting.
+   Therefore, the transformer instance given to the pipeline cannot be
+   instrospected directly. Use the attribute ``named_steps`` to introspect
+   estimators within the pipeline::
+
+     >>> from __future__ import print_function, division
+     >>> from tempfile import mkdtemp
+     >>> from shutil import rmtree
+     >>> from sklearn.datasets import load_digits
+     >>> from sklearn.pipeline import Pipeline
+     >>> from sklearn.svm import LinearSVC
+     >>> from sklearn.decomposition import PCA
+     >>> # Load some data
+     >>> digits = load_digits()
+     >>> # Create a normal pipeline
+     >>> pca1 = PCA()
+     >>> svm1 = LinearSVC()
+     >>> pipe = Pipeline([('reduce_dim', pca1), ('clf', svm1)])
+     >>> pipe.fit(digits.data, digits.target)
+     ... # doctest: +NORMALIZE_WHITESPACE, +ELLIPSIS
+     Pipeline(memory=None,
+              steps=[('reduce_dim', PCA(...)), ('clf', LinearSVC(...))])
+     >>> # The pca instance can be introspected directly
+     >>> print(pca1.components_) # doctest: +NORMALIZE_WHITESPACE, +ELLIPSIS
+         [[ -1.77484909e-19  ... 4.07058917e-18]]
+     >>> # Create a pipeline able to cache
+     >>> cachedir = mkdtemp()
+     >>> pca2 = PCA()
+     >>> svm2 = LinearSVC()
+     >>> cached_pipe = Pipeline([('reduce_dim', pca2), ('clf', svm2)], memory=cachedir)
```

Comment:
Instead of allocating local variables pca2 and svm2 here, the example would be lighter as:

```python
    >>> cached_pipe = Pipeline([('reduce_dim', PCA()),
    ...                         ('clf', LinearSVC())], memory=cachedir)
```